rts/Makefile

```diff
@@ -45,6 +46,7 @@ TOMMATH_FLAGS = \
   -DMP_MEMSET=0 \
   -DMP_FIXED_CUTOFFS \
   -DMP_NO_FILE \
+  -D__STDC_IEC_559__ \
```

Comment:
@rossberg is this a reasonable thing to define? I am not terribly informed about the float format employed in Wasm.

Reply:
Background: without this, `mp_set_double` is not defined by `LibTomMath`.

Reply:
I think we could just define `MP_HAS_SET_DOUBLE`, no? See https://github.com/libtom/libtommath/blob/44ee82cd34d0524c171ffd0da70f83bba919aa38/tommath_private.h#L167-L172

Reply:
> is this a reasonable thing to define? I am not terribly informed about the float format employed in Wasm.

Wasm uses IEEE754: https://webassembly.github.io/spec/core/syntax/types.html#number-types

Reply:
I didn't know about `MP_HAS_SET_DOUBLE`, thanks!

Reply:
Done: 89189ed5d.

Reply:
I had to reopen, as tests failed. The commit had to be reverted.

Reply:
If `__STDC_IEC_559__` is only tested in `mp_set_double.c` then we _should_ be okay, right?

Reply:
> If __STDC_IEC_559__ is only tested in mp_set_double.c then we should be okay, right?


I'm a bit confused, why do you want `__STDC_IEC_559__`? Do the tests pass with `__STDC_IEC_559__` but not `MP_HAS_SET_DOUBLE`? I don't think that's possible, `STDC_IEC_559` is only used to set `MP_HAS_SET_DOUBLE`.

Reply:
@ggreif OK, figured it out. The version I linked above is quite different than the version we're using. The version we're using needs `__STDC_IEC_559__`:

```
#if defined(__STDC_IEC_559__) || defined(__GCC_IEC_559)
mp_err mp_set_double(mp_int *a, double b) ...
```

If the code assumes IEC 559 then it won't work even if we enable it.

Reply:
I compared the implementations and they seem identical:

```diff
~ $ diff -u /nix/store/19iw4ws6xhqla72blr8w1g9x11gd4rpd-libtommath-src/bn_mp_set_double.c opt/libtommath/mp_set_double.c
--- /nix/store/19iw4ws6xhqla72blr8w1g9x11gd4rpd-libtommath-src/bn_mp_set_double.c       1970-01-01 02:00:01.000000000 +0200
+++ opt/libtommath/mp_set_double.c      2021-01-02 11:47:36.755469002 +0300
@@ -1,9 +1,9 @@
 #include "tommath_private.h"
-#ifdef BN_MP_SET_DOUBLE_C
+#ifdef MP_SET_DOUBLE_C
 /* LibTomMath, multiple-precision integer library -- Tom St Denis */
 /* SPDX-License-Identifier: Unlicense */

-#if defined(__STDC_IEC_559__) || defined(__GCC_IEC_559)
+#if defined(MP_HAS_SET_DOUBLE)
 mp_err mp_set_double(mp_int *a, double b)
 {
    uint64_t frac;
@@ -16,7 +16,7 @@
    cast.dbl = b;

    exp = (int)((unsigned)(cast.bits >> 52) & 0x7FFu);
-   frac = (cast.bits & ((1uLL << 52) - 1uLL)) | (1uLL << 52);
+   frac = (cast.bits & (((uint64_t)1 << 52) - (uint64_t)1)) | ((uint64_t)1 << 52);

    if (exp == 0x7FF) { /* +-inf, NaN */
       return MP_VAL;
@@ -30,7 +30,7 @@
       return err;
    }

-   if (((cast.bits >> 63) != 0uLL) && !MP_IS_ZERO(a)) {
+   if (((cast.bits >> 63) != 0u) && !mp_iszero(a)) {
       a->sign = MP_NEG;
    }
```

Without knowing anything about IEC 559, I'm guessing since the implementations are identical and `__STDC_IEC_559__` is not used elsewhere in the version we're using, it should be OK to define `__STDC_IEC_559__`. Alternatively we could update libtommath and define `MP_HAS_SET_DOUBLE`.

The documentation does not mention any of these macros, I'm guessing they define those in the configure script.

Reply:
Yes, I guessed so. Maybe I'll write an issue, so we don't commit errors when upgrading the `libTomMath`.

Reply:
I have filed #2739 to track this.

Reply:
Would be good to add a comment here saying that tommath's implementation works fine with IEEE 754. No idea why they called it `IEC_559`.